ledger/src/bank_forks.rs

```diff
@@ -334,10 +335,17 @@ impl BankForks {
         Ok(())
     }
 
-    fn prune_non_root(&mut self, root: Slot) {
+    fn prune_non_root(&mut self, root: Slot, largest_confirmed_root: Option<Slot>) {
         let descendants = self.descendants();
-        self.banks
-            .retain(|slot, _| slot == &root || descendants[&root].contains(slot));
+        self.banks.retain(|slot, _| {
+            slot == &root
+                || descendants[&root].contains(slot)
+                || slot < &root && slot >= &largest_confirmed_root.unwrap_or(root)
+        });
+        datapoint_info!(
```

Comment:
Do we need this as `datapoint_info!`?  Maybe `datapoint_debug!`, since this seems like it would be logged a couple times a second if a node is operating normally